proxy/client.go

```diff
@@ -69,20 +70,28 @@ func (r *remoteClientCreator) NewABCIClient() (abcicli.Client, error) {
 // DefaultClientCreator returns a default ClientCreator, which will create a
 // local client if addr is one of: 'counter', 'counter_serial', 'kvstore',
 // 'persistent_kvstore' or 'noop', otherwise - a remote client.
-func DefaultClientCreator(addr, transport, dbDir string) ClientCreator {
+//
+// The Closer is a noop except for persistent_kvstore applications,
+// which will clean up the store.
+func DefaultClientCreator(addr, transport, dbDir string) (ClientCreator, io.Closer) {
```

Comment:
what do you think about a closer function type of `func()` instead of `io.Closer`? then you do not need to create `noopCloser`

Reply:
This is backport of upstream: https://github.com/tendermint/tendermint/pull/6311/files#diff-ac53245fa7c37f12472538afe3b39784ae317e77df4e9a7624471b6e6bb8ca33R73-R76 , I don't want to change it